sptr.hh

```diff
@@ -4,151 +4,22 @@
 #ifndef __SPTR_HH_INCLUDED__
 #define __SPTR_HH_INCLUDED__
 
-// A generic non-intrusive smart-pointer template. We could use boost::, tr1::
-// or whatever, but since there's no standard solution yet, it isn't worth
-// the dependency given the simplicity of the template.
+// using std::shared_ptr
```

Comment:
This change is not necessarily a good idea. If a `sptr` is used by a single thread, the extra mutex locking in `std::shared_ptr` could be detrimental to performance. Ideally `sptr` usages should be replaced by `std::unique_ptr` usages, one by one. But such changes (and this `std::shared_ptr` change too) are completely out of scope of this pull request.

Reply:
Or was this changed because the QtWebEngine version uses `sptr` in multithreaded context, unlike the QtWebKit version?

Reply:
revert to original version.